src/e2e/sdk_actions.ts

```diff
@@ -0,0 +1,397 @@
+		// const exitProof = await session.exit();
+		//
+		// -- Wait until the current phase of Erdstall has completed!
+		// -- ^ This can be done using the `ErdstallClient.on("phaseshift", cb)`
+		// --   callback!
+		//
+		// const { stages } = await session.withdraw(exitProof);
+		const { stages } = await session.leave();
+		for await (const [_name, stage] of stages) {
+			await stage.wait();
+		}
+	},
+
+	mint: async (session: Session, nftID: bigint) => {
+		// FIXME: Is this okay to say?
+		//
+		// Offchain minting is simply a matter of passing an ERC721 contract
+		// address and an ID (unique!). Currently it is possible to mint for any
+		// contract offchain which is registered in Erdstall and counts as an
+		// `ERC721Mintable`.
+		return session.mint(PART_ADDR, nftID);
+	},
+
+	trade: async (
+		charlie: Session,
+		dagobert: Session,
+		charlieNft: bigint,
+		wantedPrnAmount: bigint,
+	) => {
+		// Charlie trades his PART-NFT for Dagobert's PRN.
+		const offer = new Assets({
+			token: PART_ADDR,
+			asset: new assets.Tokens([charlieNft]),
+		});
+		const expect = new Assets({
+			token: PERUN_ADDR,
+			asset: new assets.Amount(wantedPrnAmount),
+		});
+
+		// Creating an offer will request the user to sign it beforehand.
+		const tradeOffer = await charlie.createOffer(offer, expect);
+		charlie.once("receipt", (charlieReceipt) => {
+			// Charlie would like to know whether or not someone accepted his
+			// tradeoffer. So he listens to soft breezes of txs whizzing around
+			// Erdstall in the hopes of receiving a `Trade` receipt:
+			const tx = charlieReceipt.tx;
+			switch (tx.txType()) {
+				case Trade:
+				// Charlie lives happily ever after, because he was able to sell his
+				// basket of Monkeys.
+			}
+		});
+
+		// The interesting part is HOW this offer reaches Dagobert. This is not
+		// part of Erdstall itself but rather a peculiarity of your application.
+		// E.g. a NFT marketplace might implement its own trading logic and
+		// distribution. Erdstall is only a validation and execution layer for this
+		// matter.
+		//
+		// Here, Dagobert might have received the tradeoffer directly from Charlie,
+		// or via a third party by winning in an auction.
+		//
+		// To finalise the trade, if both parties agree of course, the acceptee has
+		// to sign the trade offer of Charlie and publish it within Erdstall, where
+		// it gets validated and executed.
+		return dagobert.acceptTrade(tradeOffer);
+	},
+
+	burn: async (dagobert: Session, formerCharlieNft: bigint) => {
+		// Burning a NFT requires being the owner of it. Since Charlie transferred
+		// his NFT in the step before, Dagobert is now able to do as he pleases.
+		// Maybe he was not a fan of monkeys and now wants some lit whales.
+		await dagobert.burn(
+			new Assets({
+				token: PART_ADDR,
+				asset: new assets.Tokens([formerCharlieNft]),
+			}),
+		);
+	},
+
+	leavingAndSeeFundsOnchain: async function (
+		session: Session,
+		provider: ethers.providers.Provider,
```

Comment:
Not yet. I mentioned in further down the tutorial why this is currently the case and that this might (most likely) change in the future. I did not put it here because the `Typing` was a pain, so I wanted to do this in a follow-up. We already have an **old** issue for that.